### PR TITLE
Feature/jvisenti/gh 20 builtin transforms

### PR DESCRIPTION
**Pull request by jvisenti**
#20 and #26 

**NOTE:** This PR breaks backwards compatibility, so will be part of a 2.0 release.

### CODE REVIEW

File: RZDataBinding/NSObject+RZDataBinding.m [hunk 1]
```diff
@@ -262,16 +263,16 @@ - (void)rz_observeBoundKeyChange:(NSDictionary *)change
     if ( boundKey != nil ) {
         id value = change[kRZDBChangeKeyNew];
 
-        [self rz_setBoundKey:boundKey withValue:value function:change[kRZDBChangeKeyBindingFunctionKey]];
+        [self rz_setBoundKey:boundKey withValue:value transform:change[kRZDBChangeKeyBindingTransformKey]];
     }
 }
 
-- (void)rz_setBoundKey:(NSString *)key withValue:(id)value function:(RZDBKeyBindingFunction)function
+- (void)rz_setBoundKey:(NSString *)key withValue:(id)value transform:(RZDBKeyBindingTransform)transform
```
Comment (KingOfBrian) [inline RZDataBinding/NSObject+RZDataBinding.m]: Weird that some have `bindingTransform`, `transform` or `withTransform` -- I'd stick with `withTransform`?

Comment (jvisenti) [inline RZDataBinding/NSObject+RZDataBinding.m]: These are private internal methods anyway, but `withValue:withTransform:` sounds weird to me. Usually there's just one `with`

File: Tests/RZDBTests/RZDBTests.m [hunk 2]
```diff
@@ -287,6 +287,48 @@ - (void)testBindingChains
...
+
+    value = nil;
+    value = kRZDBNilToCGSizeZeroTransform(value);
+    CGSize size = (CGSize){1.0f, 1.0f};
+    [value getValue:&size];
+    XCTAssertTrue(CGSizeEqualToSize(size, CGSizeZero));
+
+    value = nil;
+    value = kRZDBNilToCGRectZeroTransform(value);
+    CGRect rect = (CGRect){1.0f, 1.0f, 1.0f, 1.0f};
```
Comment (ZevEisenberg) [inline Tests/RZDBTests/RZDBTests.m]: Prefer `CGRectMake` or C struct designated initializer syntax.

File: RZDataBinding/NSObject+RZDataBinding.m (hunk 1 above)
Comment (KingOfBrian) [inline RZDataBinding/NSObject+RZDataBinding.m]: Yea, with or not with is no big deal, `bindingTransform` vs `transform` makes me think they're different.

**Comment by jvisenti** [comment]
@KingOfBrian @ZevEisenberg  You guys good with me merging this?

**Pull request merged**